src/runtime/hvx_128.ll

```diff
@@ -131,3 +131,114 @@ define weak_odr <64 x i32> @halide.hexagon.mul.vw.vuh(<64 x i32> %a, <64 x i16>
   %ab = call <64 x i32> @llvm.hexagon.V6.vcombine.128B(<32 x i32> %ab_hi, <32 x i32> %ab_lo)
   ret <64 x i32> %ab
 }
+
+; Hexagon is missing shifts for byte sized operands.
+declare <32 x i32> @llvm.hexagon.V6.vaslh.128B(<32 x i32>, i32)
+declare <32 x i32> @llvm.hexagon.V6.vasrh.128B(<32 x i32>, i32)
+declare <32 x i32> @llvm.hexagon.V6.vlsrh.128B(<32 x i32>, i32)
+declare <32 x i32> @llvm.hexagon.V6.vaslhv.128B(<32 x i32>, <32 x i32>)
+declare <32 x i32> @llvm.hexagon.V6.vasrhv.128B(<32 x i32>, <32 x i32>)
+declare <32 x i32> @llvm.hexagon.V6.vlsrhv.128B(<32 x i32>, <32 x i32>)
+declare <64 x i32> @llvm.hexagon.V6.vzb.128B(<32 x i32>)
+declare <64 x i32> @llvm.hexagon.V6.vsb.128B(<32 x i32>)
+declare <32 x i32> @llvm.hexagon.V6.vshuffeb.128B(<32 x i32>, <32 x i32>)
+
+define weak_odr <128 x i8> @halide.hexagon.shl.vub.ub(<128 x i8> %a, i8 %b) nounwind uwtable readnone alwaysinline {
+  %a_32 = bitcast <128 x i8> %a to <32 x i32>
+  %bw = zext i8 %b to i32
+  %aw = call <64 x i32> @llvm.hexagon.V6.vzb.128B(<32 x i32> %a_32)
+  %aw_lo = call <32 x i32> @llvm.hexagon.V6.lo.128B(<64 x i32> %aw)
+  %sw_lo = call <32 x i32> @llvm.hexagon.V6.vaslh.128B(<32 x i32> %aw_lo, i32 %bw)
```

Comment:
I just had an idea - rather than widen, shift, and narrow, we could simply shift 16 bit elements, and then mask off the low bits of the odd elements that "leaked" from the high bits of the even elements. Only problem is I can't think of how to compute that mask without probably more instructions than this implementation, but it would be attractive because it seems like it would reduce register usage?

It's not urgently needed, but something to think about.